src/Domain/Entity/Event.php

```diff
@@ -179,4 +180,25 @@ public function getAttachments(): array
     {
         return $this->attachments;
     }
+
+    /**
+     * @param string $status Should be TENTATIVE, CONFIRMED, or CANCELLED
+     * @return $this
+     */
+    public function setStatus(string $status): self
```

Comment:
In my mind, the status of the event should be an Enum to ensure that only values of `TENTATIVE`, `CONFIRMED` or `CANCELLED` are passed.

The property `$status` should be nullable to reset the status if needed.

Reply:
How would you like the enum implemented? `const`'s? Checking if they're one of the values and throwing a `InvalidArgumentException`?

Reply:
An enum is an object that can be type hinted. The named constructor of that enum object will ensure that it contains only valid values.

The new signature will look like this: `setStatus(Status $status): self`.

Example implementation of an enum object:

````
<?php

class Status
{
    private string $value;
    private static $instances = [];

    private function __construct(string $value)
    {
        $this->value = $value;
    }

    public static function CONFIRMED(): Status
    {
        return self::fromValue('CONFIRMED');
    }

    public static function TENTATIVE(): Status
    {
        return self::fromValue('TENTATIVE');
    }

    public static function CANCELLED(): Status
    {
        return self::fromValue('CANCELLED');
    }

    public function __toString(): string
    {
        return $this->value;
    }

    private static function fromValue(string $value): self
    {
        if (!isset(self::$instances[$value])) {
            self::$instances[$value] = new self($value);
        }

        return self::$instances[$value];
    }
}

$tentative = Status::TENTATIVE();

echo $tentative === Status::TENTATIVE() ? 'true' : 'false';
echo PHP_EOL . $tentative . PHP_EOL;
```

Reply:
It seems that implementations of the Enum is not the scope of this project. If you prefer to have Enum maybe you can consider existing `marc-mabe/php-enum` as an example or as a dependency.